source/nn/activations.lisp

```diff
@@ -102,14 +102,33 @@
   (!mul x (!sigmoid (!* (!const x 1.702) x))))
 (defmethod call ((op GeLU) &rest inputs) (gelu/call op (slot-value op 'approx) (car inputs)))
 (defun !gelu (x &key (approx :tanh)) (forward (GeLU :approx approx) x))
-;; TODO: SeLU
-;; TODO: Mish (needs tanh)
+
+(defmodel (SeLU () :where "A[~] -> A[~]") ())
+(defmethod call ((op SeLU) &rest inputs &aux (x (car inputs))) (!mul (!elu x :alpha 1.67326) (!const x 1.0507)))
+(defun !selu (x) (forward (SeLU) x))
+
+;(defmodel (Mish () :where "A[~] -> A[~]") ())
+;(defmethod call ((op Mish) &rest inputs &aux (x (car inputs))) (!mul x (!tanh (!softplus x))))
+;(defun !mish (x) (forward (Mish) x))
+
 (defmodel (HardSwish () :where "A[~] -> A[~]") ())
 (defmethod call ((op HardSwish) &rest inputs &aux (x (car inputs)))
   (!* x (!relu6 (!add x (!const x 3))) (!const x (/ 1 6))))
 (defun !hardswish (x) (forward (HardSwish) x))
-;; TODO: Hard_Tanh
-;; TODO: Softmin
+
+(defmodel (HardTanh (&key (min_val -1.0) (max_val 1.0)) :where "A[~] -> A[~]") ((min_val min_val) (max_val max_val)) )
+(defmethod call ((op HardTanh) &rest inputs)
+  (let ((x (car inputs))
+        (min_val (slot-value op 'min_val))
+        (max_val (slot-value op 'max_val)))
+    (!minimum (!maximum x (!const x min_val)) (!const x max_val))
+    ))
```

Comment:
i mean if you are not supposed to add new features, newline should not inserted here. (like other functions)